src/fun/parser.rs

```diff
@@ -1090,7 +1090,9 @@ pub trait ParserCommons<'a>: Parser<'a> {
     };
     let num_str = self.take_while(move |c| c.is_digit(radix) || c == '_');
     let num_str = num_str.chars().filter(|c| *c != '_').collect::<String>();
-    if num_str.is_empty() {
+    if let Some(c) = self.peek_one() && c.is_numeric() {
```

Comment:
Shouldn't be just `is_numeric`, but rather `is_alphanumeric` since it will include hex numbers and avoid confusions with names that start with numbers (which are not possible since numbers have higher priority).
```suggestion
    if let Some(c) = self.peek_one() && c.is_ascii_alphanumeric() {
```